extensions/amp-ad-network-fake-impl/0.1/external-reorder-head-transformer.js

```diff
@@ -0,0 +1,289 @@
+        const child = head.children.item(i);
+        switch (child.tagName) {
+          case 'META':
+            this.registerMeta_(child);
+            break;
+          case 'SCRIPT':
+            this.registerScript_(child);
+            break;
+          case 'STYLE':
+            this.registerStyle_(child);
+            break;
+          case 'LINK':
+            this.registerLink_(child);
+            break;
+          case 'NOSCRIPT':
+            this.headComponents_.noscript = child;
+            break;
+          default:
+            if (this.headComponents_.other.indexOf(child) == -1) {
+              this.headComponents_.other.push(child);
+            }
+        }
+      }
+      for (let i = 0; i < head.childNodes.length; i++) {
+        head.removeChild(head.childNodes.item(i));
+      }
+      this.repopulate_(head);
+    }
+
+    return head;
+  }
+
+  /**
+   * Classifies all elements with meta tags
+   * @param {Element} element
+   *
+   */
+  registerMeta_(element) {
+    if (element.hasAttribute('charset')) {
+      this.headComponents_.metaCharset = element;
+      return;
+    }
+    if (this.headComponents_.metaOther.indexOf(element) == -1) {
+      this.headComponents_.metaOther.push(element);
+      return;
+    }
+  }
+
+  /**
+   * Classifies all elements with script tags
+   * @param {Element} element
+   *
+   */
+  registerScript_(element) {
+    const src = element.getAttribute('src');
+    const isAsync = element.hasAttribute('async');
+    const isExtension =
+      element.hasAttribute('custom-element') ||
+      element.hasAttribute('custom-template') ||
+      element.hasAttribute('host-service');
+    if (isExtension) {
+      const custom = element.getAttribute('custom-element');
+      if (
+        custom == 'amp-story' ||
+        custom == 'amp-experiment' ||
+        custom == 'amp-dynamic-css-classes'
+      ) {
+        this.headComponents_.scriptRenderDelayingExtensions.push(element);
+        return;
+      }
+      this.headComponents_.scriptNonRenderDelayingExtensions.push(element);
+      return;
+    }
+    if (
+      isAsync &&
+      startsWith(src, urls.cdn) &&
+      (endsWith(src, '/v0.js') ||
+        endsWith(src, '/v0.js.br') ||
+        endsWith(src, '/amp4ads-v0.js') ||
+        endsWith(src, '/amp4ads-v0.js.br'))
+    ) {
+      this.headComponents_.scriptAmpEngine = element;
+      return;
+    }
+    if (
+      isAsync &&
+      startsWith(src, urls.cdn + '/v0/amp-viewer-integration-gmail-') &&
+      endsWith(src, '.js')
+    ) {
+      this.headComponents_.scriptGmailAmpViewer = element;
+      return;
+    }
+    if (
+      isAsync &&
+      (startsWith(src, urls.cdn + '/v0/amp-viewer-integration-') ||
+        (startsWith(src, urls.cdn + '/viewer/google/v') &&
+          endsWith(src, '.js')))
+    ) {
+      this.headComponents_.scriptAmpViewer = element;
+      return;
+    }
+    if (isAsync) {
+      this.headComponents_.otherScripts.push(element);
```

Comment:
is this only for inabox?

Reply:
I thought it was only for inabox, so I added a condition to check for just inabox, but that broke the visual tests. So it seems like there are other script(s) being added as well; not sure what though, because the tests show up fine on localhost.

Reply:
Found the issue; in the visual tests, the script url was being replaced by `/dist/amp4ads-v0.js` instead of `/dist/amp-inabox.js` because it was in compiled mode. I added conditions for both.